lib/ruby_lsp/document.rb

```diff
@@ -113,11 +119,12 @@ def apply_edit(source, range, text)
     class Scanner
       extend T::Sig
 
-      sig { params(source: String).void }
-      def initialize(source)
+      sig { params(source: String, encoding: String).void }
+      def initialize(source, encoding)
         @current_line = T.let(0, Integer)
         @pos = T.let(0, Integer)
         @source = source
+        @encoding = encoding
       end
 
       sig { params(position: PositionShape).returns(Integer) }
```

Comment:
It's not added in this PR, but I was confused by `find_position` taking a `position` argument and returning something that's also a position but with a different type (and perhaps meaning as well). Is it possible to give it a more clear name, like `find_char_position`?

Reply:
Renamed and added a comment for the method.